src/main/java/io/cdap/plugin/http/sink/batch/HTTPRecordWriter.java

```diff
@@ -110,8 +110,10 @@ private void executeHTTPService() throws IOException {
 
       Map<String, String> headers = config.getRequestHeadersMap();
 
-      Header authorizationHeader = config.getAuthorizationHeader();
-      if (authorizationHeader != null) {
+      AccessToken accessToken = config.getAccessToken();
```

Comment:
existing issue, but this record writer fetches a new access token for each request. This is especially bad if the batch size is 1, then it is getting a new token for every single record. It should instead reuse the token unless it's about to expire.

Reply:
Added